src/renderer/setupPassword.html

```diff
@@ -0,0 +1,222 @@
+        src="img/powered_by_holochain.png"
+      />
+      <img
+        tabindex="0"
+        id="close-btn"
+        title="close"
+        style="cursor: pointer; position: absolute; top: 5px; right: 5px; width: 35px"
+        src="img/close.svg"
+      />
+    </div>
+    <script>
+      // TODO add event listener for holochain fatal panic case
+
+      let password = '';
+      let confirmedPassword = '';
+
+      (async () => {
+        const { productName, version } = await window.electronAPI.getNameAndVersion();
+        document.getElementById('productName').innerHTML = productName;
+        document.getElementById('version').innerHTML = `version ${version}`;
+      })();
+      const activitEl = document.getElementById('activity');
+      window.electronAPI.onProgressUpdate((event, msg) => {
+        console.log('status', msg);
+        activitEl.innerHTML = msg;
+      });
+      const closeBtn = document.getElementById('close-btn');
+      closeBtn.addEventListener('click', () => {
+        window.electronAPI.exit();
+      });
+      closeBtn.addEventListener('keypress', (e) => {
+        if (e.key === 'Enter' || e.key === ' ') {
+          window.electronAPI.exit();
+        }
+      });
+
+      const choosePwDiv = document.getElementById('choose-password-div');
+      const confirmPwDiv = document.getElementById('confirm-password-div');
+      const continueBtn = document.getElementById('continue-btn');
+      const choosePwInput = document.getElementById('choose-password-input');
+      const confirmPwInput = document.getElementById('confirm-password-input');
+      const warning1El = document.getElementById('warning-1');
+      const warning2El = document.getElementById('warning-2');
+      const launchBtn = document.getElementById('launch-btn');
+      const previousStepBtn = document.getElementById('previous-step-btn');
+      const understoodDiv = document.getElementById('understood-div');
+      const understoodCheckbox = document.getElementById('understood-checkbox');
+
+      const handleLaunch = async () => {
+        if (password !== confirmedPassword) {
+          warning2El.innerHTML = "Passwords don't match!";
+          confirmPwInput.focus();
+        } else if (!understoodCheckbox.checked) {
+          understoodDiv.classList.add('red');
+        } else {
+          confirmPwDiv.classList.add('hidden');
+          activitEl.classList.remove('hidden');
+          try {
+            await window.electronAPI.launch({ type: "user-provided", password });
+          } catch (e) {
+            console.error('Failed to launch: ', e);
+            activitEl.innerHTML = '';
+            password = '';
+            choosePwInput.value = '';
+            confirmedPassword = '';
+            confirmPwInput.value = '';
+            activitEl.classList.add('hidden');
+            choosePwDiv.classList.remove('hidden');
+            warning1El.innerHTML = `ERROR: Failed to launch Holochain.`;
```

Comment:
I think providing some context too the user here would be better, also `textContent` would be much safer if context from `e` is included in the message

Reply:
I thought that providing more detail here would just clutter the UI without added value because if holochain panics (which is the most likely reason for failure here) there will be no meaningful error returned here, it is only in the log file.
Maybe I can add a remark that the actual error has been printed to the console? Eventually I would like to offer a way to export the holochain logs if that happens (see https://github.com/holochain/kangaroo-electron/issues/21) because the actual error reason is in the log file (or even in the generated .toml file whose path can be found in the log file).

Reply:
> Maybe I can add a remark that the actual error has been printed to the console? Eventually I would like to offer a way to export the holochain logs if that happens

Ahh, I see, this should be sufficient for now then, exporting the logs will also be a great addition